### PR TITLE
CASMTRIAGE-5367: Rearrange (?i) option

### PR DESCRIPTION
**Pull request by jsollom-hpe**
## Summary and Scope

The case-insensitive option (?i) needs to appear at the beginning of the regular expression line. If it does not, later versions of Python flag this as an error.

(cherry picked from commit ed6ffe7abcf2f79bb39760f56ca2af75132d9c42)

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [CASMTRIAG-5367](issue link)
* Change will also be needed in `<insert branch name here>`
* Future work required by [issue id](issue link)
* Documentation changes required in [issue id](issue link)
* Merge with/before/after `<insert PR URL here>`

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * `Crossroads24`

### Test description:

I uploaded and ran the new BOS image with the fix. I reran the triggering command, and it did not cause the BOS API to crash. Mission accomplished.

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations
Low; it's broken otherwise.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by jsl-hpe — APPROVED** [review]

**Comment by mharding-hpe** [comment]
Also, it looks like using `(?i)` at all isn't technically allowed in OpenAPI. OpenAPI docs say that the  "regular expression syntax used is from JavaScript (more specifically, [ECMA 262](https://www.ecma-international.org/ecma-262/5.1/#sec-15.10.1))". And that type of regular expression doesn't let you use case insensitivity in that way. The only way to do case insensitive stuff for format that, sadly, is the lengthy `[aA][bB][cC]` style.

**Comment by mharding-hpe** [comment]
I've opened this as an alternate PR:
https://github.com/Cray-HPE/bos/pull/134

I believe it addresses the problems raised in the triage ticket, and also corrects the additional issues that I noted.

**Comment by jsollom-hpe** [comment]
> Should we also update the CHANGELOG?

Yes, I'll do that.

File: api/openapi.yaml.in [hunk 1]
```diff
@@ -381,7 +381,7 @@ components:
           description: |
             The network over which the node will boot from.
             Choices:  NMN -- Node Management Network
-            pattern: '^(?i)nmn$'
+            pattern: '(?i)^nmn$'
```
[the remark this answers is not in the document]
Comment (jsollom-hpe) [reply api/openapi.yaml.in:384]: I accepted this fix.

**Pull request closed without merging**